assets/js/base/context/cart-checkout/payment-methods/use-payment-method-registration.js

```diff
@@ -131,7 +149,22 @@ const usePaymentMethodRegistration = (
 	return isInitialized;
 };
 
-export const usePaymentMethods = ( dispatcher ) =>
-	usePaymentMethodRegistration( dispatcher, getPaymentMethods() );
-export const useExpressPaymentMethods = ( dispatcher ) =>
-	usePaymentMethodRegistration( dispatcher, getExpressPaymentMethods() );
+export const usePaymentMethods = ( dispatcher ) => {
+	const standardMethods = getPaymentMethods();
+	// Ensure all methods are present in order.
+	// Some payment methods may not be present in PAYMENT_GATEWAY_SORT_ORDER if they
+	// depend on state, e.g. COD can depend on shipping method.
+	const displayOrder = union(
```

Comment:
I didn't test it, but just leaving the suggestion here in case you think it makes sense: if you want to avoid using lodash, you can use a native [Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) here. It will ensure items are not duplicated and they are in order.

```JSX
const displayOrder = new Set( [
	...PAYMENT_GATEWAY_SORT_ORDER,
	...Object.keys( standardMethods )
] );
```

Then, you will need to replace the [line above](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/2934/files#diff-71ebb9a05002d896e445bb08e1294645R98) (`for ( let i = 0; i < paymentMethodsSortOrder.length; i++ ) { ... }`) with `paymentMethodsSortOrder.forEach( ... )`.

As I said, not a blocker at all. :slightly_smiling_face: